File: presentation/src/main/java/cleanarcpro/brightowusu/com/cleanarcproj/di/modules/AppModule.java
package cleanarcpro.brightowusu.com.cleanarcproj.di.modules;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module//(includes = {RepositoryModule.class, NetworkModule.class, InteractorsModule.class})
public class AppModule {

    public Context myApplication;

    public AppModule(Context myApplication) {
        this.myApplication = myApplication;
    }

    @Provides
    @Singleton
    public Context providesApplication() {
        return myApplication;
    }
}

File: presentation/src/main/java/cleanarcpro/brightowusu/com/cleanarcproj/di/modules/AppModule.kt
package cleanarcpro.brightowusu.com.cleanarcproj.di.modules;

import android.content.Context

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
class AppModule(val myApplication : Context) {


    @Provides
    @Singleton
    fun providesApplication(): Context {
        return myApplication
    }
}
